Trevorgb gi2149 update times

trevorgb opened the pull request:
implements fixes for issue 1859
implements fixes for issue 2149

Supersedes pull request 835.

trevorgb committed:
GoogleIssue 1859
Addresses issue with hardcoded time for full updates.

Command line option added
 - u <time>
 --updatetime=<time>

Accepts an integer between 0 and 23 to specify the hour to update at.

trevorgb committed:
adds interface option on CONFIG/GENERAL to set the update time

thezoggy commented on sickbeard/showUpdater.py:
```diff
@@ -36,8 +37,9 @@ def __init__(self):
-        run_updater_time = datetime.time(hour=3)
+        # update at specified time.
+        run_updater_time = datetime.time(hour=sickbeard.UPDATE_TIME)
+        sys.exit(str(run_updater_time))
```
leaving some debugging code in?

thezoggy commented on sickbeard/__init__.py:
```diff
@@ -349,7 +350,7 @@ def initialize(consoleLogging=True):
                 GIT_PATH, MOVE_ASSOCIATED_FILES, \
                 COMING_EPS_LAYOUT, COMING_EPS_SORT, COMING_EPS_DISPLAY_PAUSED, METADATA_WDTV, METADATA_TIVO, IGNORE_WORDS, CREATE_MISSING_SHOW_DIRS, \
-                ADD_SHOWS_WO_DIR, ANON_REDIRECT
+                ADD_SHOWS_WO_DIR, ANON_REDIRECT, UPDATETIME
```
your variable is `UPDATE_TIME` NOT UPDATETIME.

also your not setting/checking if the variable exists.. 
`UPDATE_TIME = bool(check_setting_int(CFG, 'General', 'update_time', 3))`

update_time also is a bad variable.. i have no idea the context...

thezoggy commented on SickBeard.py:
```diff
@@ -234,6 +235,13 @@ def main():
 
+        # Do the daily full update at a different time.
+        if o in ('-u', '--updatetime='):
+            enteredTime = int(a)
```
what if its not an int.. prob would break.. look how the --port option works
https://github.com/midgetspy/Sick-Beard/blob/development/SickBeard.py#L258

trevorgb committed:
corrects typo in sickbeard/_init_.py
removes leftover debug code in sickbeard/showUpdater.py
adds error trapping for non-numbers for the -u or --updatetime values
